### ISSUE TITLE
[API] - Supplier Invoice Insert/Update Mutation

### ISSUE THREAD

**Issue by andreievg** [openmsupply/remote-server#349]
MW parent job: `[OMSBEA] omSupply Back End - API`

### Description

Mutations for insert and update of supplier invoice (excluding lines)

Should be straight forward since: https://github.com/openmsupply/remote-server/issues/348 would already be done

### Scope

<ins>Includes</ins>
* Invoice table changes (if not done yet with customer invoice)
* Supplier Invoice Insert Mutation
* Supplier Invoice Update Mutation
* When status changes update dates
* Validations
  * Name is Supplier and is Provided
  * For insert, ID doesn't already exist
  * For update, ID exists
  * Can't change status to draft once confirmed
  * Cannot edit at all once finalised

<ins>Excludes</ins>
* Customer Invoice lines

<ins>Blockers</ins>
* How to do business logic ?
* Not a blocked for dev but for completion -> Invoice API

### Tasks

- [x] ~~[change schema as per translations](https://github.com/openmsupply/remote-server/issues/356)~~
- [x] ~~[insert mutation](https://github.com/openmsupply/remote-server/issues/364)~~
- [x] ~~[update mutation](https://github.com/openmsupply/remote-server/issues/365)~~
- [x] Types https://github.com/openmsupply/remote-server/issues/416
- [x] Insert https://github.com/openmsupply/remote-server/issues/417
- [x] Update https://github.com/openmsupply/remote-server/issues/418
- [x] Delete https://github.com/openmsupply/remote-server/issues/419
- [ ] ~~add tests~~ new issue
- [x] MW job added

**Comment by andreievg** [comment]
I haven't seen this one being assigned yet, so just assigned to self, I think it's important to move it along

**Comment by wlthomson** [comment]
Probably better that I don't work on #348 in parallel as I think that would be inefficient?
